### PR TITLE
feat(core-cell-renderer): added duration cell renderer

### PR DESCRIPTION
**Pull request by 23nobody**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import 'font';
+
+.duration-cell {
+    .duration-text{
+        @include ellipsis-overflow();
+        @include body-1-regular();
+    }
+}
```
Comment (arjunlalb) [inline projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.scss:8]: 2 space indentation here please

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @Inject(TABLE_ROW_DATA) rowData: TableRow
+  ) {
+    super(columnConfig, index, parser, cellData, rowData);
+  }
+
+  public ngOnInit(): void {
+    super.ngOnInit();
+
+    this.durationText = this.value.toMultiUnitString(TimeUnit.Second, true, UnitStringType.Long)
+  }
```
Comment (arjunlalb) [inline projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts:45]: you can use a method here with memoization and avoid defining constructor and init blocks.

Comment (aaron-steinfeld) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts:45]: Better yet, just use the duration pipe and avoid both (explicit) memoization and the code. If we're looking for different behavior for the formatting, either the pipe should be updated, or we want a custom renderer. In other words, the developer expectation should be the duration cell renderer and duration pipe behave the same way.

Comment (arjunlalb) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts:45]: good point. I'd say, modify the pipe to also support long format display and use it here.

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.scss (hunk 1 above)
Comment (23nobody) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.scss]: updated

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { CoreTableCellParserType } from '../../types/core-table-cell-parser-type';
+import { CoreTableCellRendererType } from '../../types/core-table-cell-renderer-type';
+import { TableCellAlignmentType } from '../../types/table-cell-alignment-type';
+
+@Component({
+  selector: 'ht-duration-table-cell-renderer',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="duration-cell">
+    <span class="duration-text" [htTooltip]="this.durationText">{{ this.durationText | htDisplayString }}</span>
```
[the remark this answers is not in the document]
Comment (23nobody) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts]: updated

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts (hunk 2 above)
Comment (23nobody) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts]: using memoization for now, have added a todo to modify pipe (as that can also affect other places). Will modify the pipe in follow-up.

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { TableCellRenderer } from '../../table-cell-renderer';
+import { TableCellRendererBase } from '../../table-cell-renderer-base';
+import { CoreTableCellParserType } from '../../types/core-table-cell-parser-type';
+import { CoreTableCellRendererType } from '../../types/core-table-cell-renderer-type';
+import { TableCellAlignmentType } from '../../types/table-cell-alignment-type';
+
+@Component({
+  selector: 'ht-duration-table-cell-renderer',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  // TODO: Modify the htDuration pipe to also support long format display and use it here.
```
Comment (arjunlalb) [inline projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts:12]: lets fix it right away. its a small change. Update htDuration pipe to take a format input.

File: projects/common/src/utilities/formatters/string/highlight.pipe.ts [hunk 5]
```diff
@@ -2,6 +2,7 @@ import { Pipe, PipeTransform } from '@angular/core';
 import { isArray } from 'lodash-es';
 import { assertUnreachable } from '../../lang/lang-utils';
 
+// TODO: Currently htHighlight does not accommodate special characters.
```
Comment (arjunlalb) [inline projects/common/src/utilities/formatters/string/highlight.pipe.ts:5]: this is the wrong description. it does not escape reserved regex characters is the issue.

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
 })
 export class DisplayDurationPipe implements PipeTransform {
-  public transform(millis?: number): string {
-    return durationFormatter(millis);
+  public transform(millis?: number, unitStringType: UnitStringType = UnitStringType.Short): string {
+    if(millis === undefined)
+    return '-';
+
+    if(unitStringType === UnitStringType.Short)
+    return new TimeDuration(millis!, TimeUnit.Millisecond).toString();
```
Comment (arjunlalb) [inline projects/common/src/utilities/formatters/duration/display-duration.pipe.ts:15]: this is not the same as the output from durationFormatter. Let's not break existing cases. continue to use `durationFormatter(millis)` for short.

File: projects/common/src/utilities/formatters/string/highlight.pipe.ts (hunk 5 above)
Comment (23nobody) [reply projects/common/src/utilities/formatters/string/highlight.pipe.ts]: updated

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts (hunk 6 above)
Comment (23nobody) [reply projects/common/src/utilities/formatters/duration/display-duration.pipe.ts]: reverted

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts (hunk 2 above)
Comment (23nobody) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts]: updated pipe

File: projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts (hunk 4 above)
Comment (23nobody) [reply projects/components/src/table/cells/data-renderers/duration/duration-table-cell-renderer.component.ts]: updated pipe

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts [hunk 7]
```diff
@@ -1,11 +1,21 @@
...
 import { durationFormatter } from './duration-formatter';
 
+
 @Pipe({
   name: 'htDisplayDuration'
 })
 export class DisplayDurationPipe implements PipeTransform {
-  public transform(millis?: number): string {
+  public transform(millis?: number, unitStringType: UnitStringType = UnitStringType.Short): string {
+  if(millis === undefined){
```
Comment (arjunlalb) [inline projects/common/src/utilities/formatters/duration/display-duration.pipe.ts:11]: there are code style fixes needed here.

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts [hunk 8]
```diff
@@ -1,11 +1,21 @@
...
+    return '-';
+  }
+
+  if(unitStringType === UnitStringType.Short){
```
Comment (arjunlalb) [inline projects/common/src/utilities/formatters/duration/display-duration.pipe.ts:15]: code style fixes here as well

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts (hunk 7 above)
Comment (23nobody) [reply projects/common/src/utilities/formatters/duration/display-duration.pipe.ts]: done

File: projects/common/src/utilities/formatters/duration/display-duration.pipe.ts (hunk 8 above)
Comment (23nobody) [reply projects/common/src/utilities/formatters/duration/display-duration.pipe.ts]: done

**Review by arjunlalb — APPROVED** [review]

**Pull request merged**